algolia/search/dictionary_entry_compound.go

```diff
@@ -0,0 +1,51 @@
+package search
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// Compound represents an entry in the Compounds dictionary.
+type Compound struct {
+	objectID      string
+	language      string
+	Word          string
```

Comment:
why are the words above with a lower case and the last ones with uppercase (for my personal information)?

Reply:
The case defines the visibility: upper case is for public, lower case is for the internal usage inside the package. 

I want all the entries to conform to `DictionaryEntry` interface, so I declared `ObjectID()` and `Language()` public functions (getters) for each of them. 

A structure cannot have a public variable and a public function with the same name, that's why they are declared with lower case.

Inspired by Anthony's synonyms implementation.

Reply:
okay I got it! thanks!